pbsmetrics/go_metrics.go

```diff
@@ -562,6 +584,15 @@ func (me *Metrics) RecordTimeoutNotice(success bool) {
 	return
 }
 
+func (me *Metrics) RecordTCFReq(version int) {
+	if version >= 0 && version < len(me.TCFVersions) {
```

Comment:
Could this be simplified / future proofed with a map lookup?:
```
if m, ok := me.TCFVersions[version]; ok {
  m.Mark(1)
  return
}
me.TCFReqVersion[TCFVersionErr].Mark(1)
return
```

Reply:
Oh, I see ... it does push some logic to the map creation. This would require we set up a mapping of version numbers to version keys, something we don't do anywhere else in metrics. All that sort of mapping is done in the PBS core code itself, with the keys then passed to the metrics code. I think we would need to run through several TCF version iterations before we would really have to worry about saving space for omitting the older versions, at which point metrics itself is probably ripe for a refactor anyway after all those years more of development.

Reply:
Discussed in person. I misread the code. The key was pulled out to a separate variable it make it easier to follow.